pkg/standalone-utils/contracts/relayer/ERC4626Wrapping.sol

```diff
@@ -0,0 +1,87 @@
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/Address.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/misc/IERC4626.sol";
+
+import "@balancer-labs/v2-vault/contracts/interfaces/IVault.sol";
+
+import "../interfaces/IBaseRelayerLibrary.sol";
+
+/**
+ * @title ERC4626Wrapping
+ * @notice Allows users to wrap and unwrap ERC4626 tokens
+ * @dev All functions must be payable so they can be called from a multicall involving ETH
+ */
+abstract contract ERC4626Wrapping is IBaseRelayerLibrary {
+    using Address for address payable;
+
+    function wrapToken(
+        IERC4626 wrappedToken,
+        address sender,
+        address recipient,
+        uint256 amount,
+        uint256 outputReference
+    ) external payable {
+        if (_isChainedReference(amount)) {
+            amount = _getChainedReferenceValue(amount);
+        }
+
+        IERC20 dynamicToken = IERC20(wrappedToken.asset());
+
+        // The wrap caller is the implicit sender of tokens, so if the goal is for the tokens
+        // to be sourced from outside the relayer, we must first pull them here.
+        if (sender != address(this)) {
+            require(sender == msg.sender, "Incorrect sender");
+            _pullToken(sender, dynamicToken, amount);
+        }
+
+        dynamicToken.approve(address(wrappedToken), amount);
+        // Use 0 for the referral code
+        uint256 result = wrappedToken.deposit(amount, recipient);
+
+        if (_isChainedReference(outputReference)) {
+            _setChainedReferenceValue(outputReference, result);
+        }
+    }
+
+    function unwrapToken(
+        IERC4626 wrappedToken,
+        address sender,
+        address recipient,
+        uint256 amount,
+        uint256 outputReference
+    ) external payable {
+        if (_isChainedReference(amount)) {
+            amount = _getChainedReferenceValue(amount);
+        }
+
+        // The unwrap caller is the implicit sender of tokens, so if the goal is for the tokens
+        // to be sourced from outside the relayer, we must first pull them here.
+        if (sender != address(this)) {
+            require(sender == msg.sender, "Incorrect sender");
+            _pullToken(sender, wrappedToken, amount);
+        }
+
+        // No approval is needed here, as the Static Tokens are burned directly from the relayer's account
+        uint256 result = wrappedToken.redeem(amount, recipient, sender);
```

Comment:
```suggestion
        uint256 result = wrappedToken.redeem(amount, recipient, address(this));
```

This will either revert or strand user's tokens on the relayer if called with the user as the sender. This is currently pulling tokens from the user onto the relayer and then attempting to redeem tokens which are still on the user's address.